lib/src/util/io.dart

```diff
@@ -130,8 +130,12 @@ void warn(String message, {bool color}) {
 /// applications that don't print out the bound port.
 Future<T> getUnusedPort<T>(FutureOr<T> tryPort(int port)) async {
   T value;
+  var supportsIPv6 = false;
+  if (await _supportsIPv6()) {
+    supportsIPv6 = true;
+  }
```

Comment:
This could just be `var supportsIPv6 = await _supportsIPv6`, but it seems perverse to bind a socket here when we're already doing so in `getUnsafeUnusedPort()`. Maybe `getUnsafeUnusedPort()` should try to bind to IPv6 first and record if that doesn't work.